src/components/Common/Environments/CreateEnvironment.vue

```diff
@@ -102,60 +108,88 @@
             </b-col>
             <span
               class="CreateEnvironment-box-feedback text-danger ml-2"
-              v-if="!colorState"
+              v-if="colorState === false"
             >
-              <small> You must select an environment color</small></span
+              <small>You must select a color for this connection</small></span
             >
           </b-row>
         </b-col>
       </b-row>
-      <b-alert :show="errors.environmentAlreadyExists" variant="danger">
-        An environment with the same name exists already
-      </b-alert>
-      <b-alert :show="errors.host" variant="danger">
-        The hostname is invalid
-      </b-alert>
-      <b-alert :show="errors.name" variant="danger">
-        The hostname is invalid
-      </b-alert>
     </b-form>
   </div>
 </template>
 
 <script>
-import { DEFAULT_COLOR, envColors } from '../../../vuex/modules/kuzzle/store'
+import { validationMixin } from 'vuelidate'
+import { numeric, required } from 'vuelidate/lib/validators'
+import { isValidHostname } from '../../../utils'
+
+import { envColors, DEFAULT_COLOR } from '../../../vuex/modules/kuzzle/store'
 const useHttps = window.location.protocol === 'https:'
+function nameIsUnique(value) {
```

Comment:
Why declare these methods here? Can't we put them in the `methods` object?

Reply:
1. these little functions are meant to be generic and pure
2. it feels wonky to declare your `validations` object by referencing `this.function`, since you don't really know if Vue first declares the methods or the validation system
3. vuelidate dynamically changes the closure around the custom validators when they are called. If the component scope is needed, it will be dynamically injected.